tools/build-templated-pages/src/features.rs

```diff
@@ -58,53 +38,88 @@ fn parse_features(panic_on_missing: bool) -> Vec<Feature> {
             if key == "default" {
                 None
             } else {
-                let name = key
-                    .as_repr()
-                    .unwrap()
-                    .as_raw()
-                    .as_str()
-                    .unwrap()
-                    .to_string();
-                if let Some(description) = key.leaf_decor().prefix() {
-                    let description = description.as_str().unwrap().to_string();
-                    if !description.starts_with("\n# ") || !description.ends_with('\n') {
-                        panic!("Missing description for feature {name}");
+                let key_name = get_key_name(key);
+                let key_prefix = key.leaf_decor().prefix();
+                if what_to_run.contains(Command::CHECK_MISSING) {
+                    if let Some(description) = key_prefix {
+                        create_feature(&default, key_name, description)
+                    } else {
+                        panic!("Missing description for feature {key_name}");
                     }
-                    let description = description
-                        .strip_prefix("\n# ")
-                        .unwrap()
-                        .strip_suffix('\n')
-                        .unwrap()
-                        .to_string();
-                    Some(Feature {
-                        is_default: default.contains(&name),
-                        name,
-                        description,
-                    })
-                } else if panic_on_missing {
-                    panic!("Missing description for feature {name}");
                 } else {
-                    None
+                    let feature = |description| create_feature(&default, key_name, description);
+                    key_prefix.map(feature).unwrap()
                 }
             }
         })
         .collect()
 }
 
+fn get_manifest_features() -> Table {
+    let manifest_file = std::fs::read_to_string("Cargo.toml").unwrap();
+    let manifest = manifest_file.parse::<Document>().unwrap();
+    let features = manifest["features"].as_table().unwrap().clone();
+    features
+}
+
+fn create_feature(default: &[String], name: String, description: &RawString) -> Option<Feature> {
```

Comment:
```suggestion
fn create_feature(default: &[&str], name: &str, description: &str) -> Option<Feature> {
```

Same as before. Our goal is to reduce allocations by passing references to existing strings.